basics_of_sst-core/tutorial/Example04.md

```diff
@@ -0,0 +1,488 @@
+            ExampleSubComponent(Component *owningComponent) :
+                SubComponent(owningComponent)
+            { }
+            ExampleSubComponent(Component* comp, Params& params) :
+                SubComponent(comp)
+            { }
+
+            // Updated constructors.  No longer pass in a parent.
+            // Instead, use a component id.
+            //
+            ExampleSubComponent(ComponentId_t id) :
+                SubComponent(id)
+            { }
+            ExampleSubComponent(ComponentId_t id, Params& params) :
+                SubComponent(id)
+            { }
+
+            virtual ~ExampleSubComponent() {}
+            
+            // Clock handler.  This is the method called from the clock event.
+            //
+            virtual void clock(Cycle_t) {}
+
+            // Subcomponent public method for returning current count.
+            //
+            int getCount() { return count_++; }
+```
+
+As is the case for components and events, all subcomponents are derived from SST::SubComponent or one of its children.
+
+The constructor and destructor are a standard format and required to interface with the SST core functions.  
+
+Subcomponents still have clock handlers but they are not automatically called by the simulation.  Instead they are called from the subcomponent parent as part of the parent's clock event handling.
+
+Finally, `getCount()` is the user method that provides the subcomponent functionality.  It simply increments and returns the count.
+
+The subcomponent is described to the SST core using SST_ELI macros.
+
+```
+            // Register the subcomponent API.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT_API(Example04::ExampleSubComponent)
+
+            // Register the subcomponent.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT_DERIVED(
+                ExampleSubComponent,                // Name of the class being registered.
+                                                    // INSERT_CLASS_NAME
+                "example",                          // Element library.  Name of the .so.
+                "ExampleSubComponent",              // Name used to reference the subcomponent.
+                                                    // Can be anything but typically set to
+                                                    // the class name.
+                SST_ELI_ELEMENT_VERSION(1, 0, 0),   // 
+                "Example subcomponent",             // Brief subcomponent description
+                Example04::ExampleSubComponent      // Name of the subcomponent interface the
+                                                    // subcomponent inherits from.  Should be
+                                                    // the full parent class name.
+                                                    // "INSERT_FULL_PARENT_CLASS_NAME" or 
+                                                    // "INSERT_COMPLETE_NAMESPACE::INSERT_PARENT_CLASS_NAME"
```

Comment:
Not necessarily the parent class name, rather the API that this subcomponent implements.